06/Main.py

```diff
@@ -0,0 +1,86 @@
+"""This file is part of nand2tetris, as taught in The Hebrew University,
+and was written by Aviv Yaish according to the specifications given in  
+https://www.nand2tetris.org (Shimon Schocken and Noam Nisan, 2017)
+and as allowed by the Creative Common Attribution-NonCommercial-ShareAlike 3.0 
+Unported License (https://creativecommons.org/licenses/by-nc-sa/3.0/).
+"""
+import os
+import sys
+import typing
+from SymbolTable import SymbolTable
+from Parser import Parser
+from Code import Code
+from tables import A_COMMAND, C_COMMAND, L_COMMAND
+
+
+def assemble_file(
+        input_file: typing.TextIO, output_file: typing.TextIO) -> None:
+    """Assembles a single file.
+
+    Args:
+        input_file (typing.TextIO): the file to assemble.
+        output_file (typing.TextIO): writes all output to this file.
+    """
+    symbol_table = SymbolTable()
+
+    parser = Parser(input_file)
+    rom_address = 0
+
+    first_iteration = True
+    while parser.has_more_commands():
+        if first_iteration:
+            first_iteration = False
+        else:
+            parser.advance()
+
+        if parser.command_type() == L_COMMAND:
+            symbol_table.add_entry(parser.symbol(), rom_address)
+        else:
+            rom_address += 1
+
+    input_file.seek(0)
+    parser = Parser(input_file)
+    translated_lines = []
+
+    first_iteration = True
```

Comment:
what?

Reply:
first iteration of loop.
renamed and added comments.